modules/nf-core/pilon/main.nf

```diff
@@ -0,0 +1,43 @@
+process PILON {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::pilon=1.24"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/pilon:1.24--hdfd78af_0':
+        'quay.io/biocontainers/pilon:1.24--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    tuple val(meta_bam), path(bam), path(bai)
```

Comment:
Because fasta, bam and bai have to come all from the same sample, would it be rather:
```suggestion
    tuple val(meta), path(fasta), path(bam), path(bai)
```
I am not sure, thats just a question because I do not know better.

Reply:
That is an interesting suggestion. Both would work.
I prefer to have them split, and it works if meta and meta_bam are the same.
In the case of purge_dups, they concatenated all the inputs and I had to split them because sometimes my meta reflects the input sequencing type (SR, PacBio, ONT), not only the sample name.

Reply:
> I prefer to have them split, and it works if meta and meta_bam are the same.

Hm ok, I would prefer having them not split. Because I think the typical application is that before the module is some channel magic that joins the required input (genome = fasta, alignment = bam & bai) to have matching files by ensuring that meta is matching.

> In the case of purge_dups, they concatenated all the inputs and I had to split them because sometimes my meta reflects the input sequencing type (SR, PacBio, ONT), not only the sample name.

That seems to me rather the exception than the usual module output. But I might be wrong.

Unfortunately the nf-core modules guidelines do not specify this clearly, see https://nf-co.re/developers/modules#inputoutput-options, i.e.  `Directly associated auxiliary files to an input file MAY be defined within the same input channel alongside the main input channel (e.g. BAM and BAI).`, so it seems I shouldn't insist on it!